containers/conformance.md

```diff
@@ -0,0 +1,235 @@
+# Zowe Conformance Criteria Related to Containerization
+
+**DRAFT**
+
+These conformance criteria are applicable for all Zowe components intending to run in a containerized environment. The containerized environment could be Kubernetes or OpenShift running on Linux or Linux on Z. This may also apply to `docker-compose` running on Linux, Windows, Mac OS, or zCX.
+
+## Image
+
+In general, the image should follow [Best practices for writing Dockerfiles](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/). Below requirements are in addition to the list.
+
+### Base Image
+
+You are free to choose a base image based on your requirements.
+
+Here are our recommendations for core Zowe components:
+
+- Zowe base images:
+  * base on Ubuntu and Red Hat Universal Base Image,
+  * provide common dependencies including JDK and/or node.js,
+  * support both `amd64` and `s390x` architecture.
+- [Red Hat Universal Base Image 8 Minimal](https://developers.redhat.com/articles/ubi-faq?redirect_fragment=resources#ubi_details)
+- [Ubuntu](https://hub.docker.com/_/ubuntu)
+
+The image should contain as few software packages as possible for security, and should be as small as possible such as by reducing package count and layers.
+
+### Multi-CPU Architecture
+
+- Zowe core components must release images based on both `amd64` and `s390x` CPU architecture.
+- Zowe core component images must use multiple manifests to define if the image supports multiple CPU architecture.
+
+### Image Label
+
+These descriptive labels are required in the Dockerfile: `name`, `maintainer`, `vendor`, `version`, `release`, `summary`, and `description`.
+
+Example line:
+
+```
+### Required Labels 
+LABEL name="APPLICATION NAME" \
+      maintainer="EMAIL@ADDRESS" \
+      vendor="COMPANY NAME" \
+      version="VERSION NUMBER" \
+      release="RELEASE NUMBER" \
+      summary="APPLICATION SUMMARY" \
+      description="APPLICATION DESCRIPTION" \
+```
```

Comment:
Do we need to provide the actual Dockerfile? As we don't actually use it per se. We would need to grab it from JIB.

Reply:
We do need the Dockerfile(s) used to build components, it will be useful to inspect license, generate sources image, or do some scans. But for Dockerfile of base images, we don't really need.

Do you mean you don't have the Dockerfile if you are using JIB to create the images?

Reply:
We don't have one per se. We can provide one if needed as this is basically the Docker image:
https://github.com/GoogleContainerTools/jib/blob/master/docs/faq.md#what-would-a-dockerfile-for-a-jib-built-image-look-like

Reply:
I see. This rule comes from one of the RedHat Certification criteria, https://github.com/zowe/zowe-install-packaging/issues/2170#issuecomment-857948519. It's unclear if Dockerfile is required or not, maybe labels in image is also acceptable.

This rule isn't critical for starting APIML in Zowe context. So if JIB can fit in other conformance rules listed here, we can change the rule to `These descriptive labels are required in the Dockerfile or image:`.